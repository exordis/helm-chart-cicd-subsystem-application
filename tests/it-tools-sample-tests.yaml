suite: Samples/IT Tools
templates:
  - templates/*.yaml
values:
  - "../docs/snippets/samples/it-tools.yaml"
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}
