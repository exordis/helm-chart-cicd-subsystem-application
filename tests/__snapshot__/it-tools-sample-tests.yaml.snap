manifest should match snapshot:
  1: |
    apiVersion: v1
    binaryData: {}
    data:
      EXORDIS_APPLICATION: samples-it-tools
      EXORDIS_ENVIRONMENT: prod
      EXORDIS_INSTANCE: ""
      EXORDIS_PRODUCT: cicd
      EXORDIS_SUBSYSTEM: samples
    kind: ConfigMap
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: samples-it-tools
        app.kubernetes.io/instance: ""
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: samples-it-tools
        app.kubernetes.io/part-of: samples
        app.kubernetes.io/version: 2024.10.22-7ca5933
        exordis/application: samples-it-tools
        exordis/application-instance: ""
        exordis/application-type: service
        exordis/environment: prod
        exordis/product: cicd
        exordis/subsystem: samples
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: samples-it-tools-envs
      namespace: samples-prod
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: samples-it-tools
        app.kubernetes.io/instance: ""
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: samples-it-tools
        app.kubernetes.io/part-of: samples
        app.kubernetes.io/version: 2024.10.22-7ca5933
        exordis/application: samples-it-tools
        exordis/application-instance: ""
        exordis/application-type: service
        exordis/application-workload: "true"
        exordis/environment: prod
        exordis/product: cicd
        exordis/subsystem: samples
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: samples-it-tools
      namespace: samples-prod
    spec:
      replicas: 3
      revisionHistoryLimit: 2
      selector:
        matchLabels:
          exordis/application: samples-it-tools
          exordis/application-instance: ""
          exordis/application-workload: "true"
          exordis/environment: prod
          exordis/subsystem: samples
      template:
        metadata:
          annotations:
            checksum/configMap.envs: 1c7b4d198a6e4256745d103b3fa4de5100ff37e813ad454caad80cd9c5a5f52e
          labels:
            app.kubernetes.io/component: samples-it-tools
            app.kubernetes.io/instance: ""
            app.kubernetes.io/managed-by: helm
            app.kubernetes.io/name: samples-it-tools
            app.kubernetes.io/part-of: samples
            app.kubernetes.io/version: 2024.10.22-7ca5933
            exordis/application: samples-it-tools
            exordis/application-instance: ""
            exordis/application-type: service
            exordis/application-workload: "true"
            exordis/environment: prod
            exordis/product: cicd
            exordis/subsystem: samples
            helm.sh/chart: cicd-subsystem-application-0.1.0
        spec:
          containers:
            - envFrom:
                - configMapRef:
                    name: samples-it-tools-envs
              image: docker.io/corentinth/it-tools:2024.10.22-7ca5933
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 30
                periodSeconds: 10
                successThreshold: 1
                tcpSocket:
                  port: 80
                timeoutSeconds: 1
              name: application
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                failureThreshold: 30
                periodSeconds: 10
                successThreshold: 1
                tcpSocket:
                  port: 80
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              startupProbe:
                failureThreshold: 30
                periodSeconds: 5
                successThreshold: 1
                tcpSocket:
                  port: 80
                timeoutSeconds: 1
              volumeMounts: []
          initContainers: null
          volumes: null
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/tls-acme: "true"
        nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
      labels:
        app.kubernetes.io/component: samples-it-tools
        app.kubernetes.io/instance: ""
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: samples-it-tools
        app.kubernetes.io/part-of: samples
        app.kubernetes.io/version: 2024.10.22-7ca5933
        exordis/application: samples-it-tools
        exordis/application-instance: ""
        exordis/application-type: service
        exordis/environment: prod
        exordis/product: cicd
        exordis/subsystem: samples
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: samples-it-tools-ui
      namespace: samples-prod
    spec:
      rules:
        - host: it-tools.local
          http:
            paths:
              - backend:
                  service:
                    name: samples-it-tools-ui
                    port:
                      name: http
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - it-tools.local
          secretName: samples-it-tools-ui-tls
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: samples-it-tools
        app.kubernetes.io/instance: ""
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: samples-it-tools
        app.kubernetes.io/part-of: samples
        app.kubernetes.io/version: 2024.10.22-7ca5933
        exordis/application: samples-it-tools
        exordis/application-instance: ""
        exordis/application-type: service
        exordis/environment: prod
        exordis/product: cicd
        exordis/service-name: samples-it-tools-ui
        exordis/subsystem: samples
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: samples-it-tools-ui
      namespace: samples-prod
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        exordis/application: samples-it-tools
        exordis/application-instance: ""
        exordis/application-workload: "true"
        exordis/environment: prod
        exordis/subsystem: samples
      sessionAffinity: None
      type: ClusterIP
